### PR TITLE
build: migrate to nimbus

### LINKED ISSUE #265: Can not build in project with 0.12.1
I clone project and run yarn install successfully, but yarn build failed.
```sh
yarn run v1.21.0
$ yarn run build:cjs && yarn run build:esm && yarn run type:dts && yarn run build:assets
$ NODE_ENV=production beemo babel --extensions=".js,.jsx,.ts,.tsx" ./src --out-dir lib/ --delete-dir-on-start --minify --workspaces="@superset-ui/!(demo|generator-superset)"
[1/2] CONFIG Creating config files
[2/2] DRIVER Running Babel v7.6.4 driver

Successfully compiled 17 files with Babel.
Successfully compiled 14 files with Babel.
Successfully compiled 10 files with Babel.
Successfully compiled 11 files with Babel.
Successfully compiled 4 files with Babel.
Successfully compiled 10 files with Babel.
Successfully compiled 17 files with Babel.
Successfully compiled 22 files with Babel.
Successfully compiled 20 files with Babel.
Successfully compiled 5 files with Babel.
Successfully compiled 18 files with Babel.

🤖  Powered by Beemo v1.1.1
$ NODE_ENV=production beemo babel --extensions=".js,.jsx,.ts,.tsx" ./src --out-dir esm/ --delete-dir-on-start --esm --minify --workspaces="@superset-ui/!(demo|generator-superset)"
[1/2] CONFIG Creating config files
[2/2] DRIVER Running Babel v7.6.4 driver

Successfully compiled 17 files with Babel.
Successfully compiled 14 files with Babel.
Successfully compiled 10 files with Babel.
Successfully compiled 11 files with Babel.
Successfully compiled 4 files with Babel.
Successfully compiled 10 files with Babel.
Successfully compiled 17 files with Babel.
Successfully compiled 22 files with Babel.
Successfully compiled 20 files with Babel.
Successfully compiled 5 files with Babel.
Successfully compiled 18 files with Babel.

🤖  Powered by Beemo v1.1.1
$ NODE_ENV=production beemo typescript --workspaces="@superset-ui/!(demo|generator-superset)" --emitDeclarationOnly
[1/2] CONFIG Creating config files (failed)

The "to" argument must be of type string. Received type object
TypeError [ERR_INVALID_ARG_TYPE]: The "to" argument must be of type string. Received type object
    at validateString (internal/validators.js:125:11)
    at Object.relative (path.js:1163:5)
    at context.workspaces.forEach.wsPath (/Volumes/Transcend/Apache/superset-ui/node_modules/@superset-ui/build-config/configs/typescript.js:24:33)
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/Volumes/Transcend/Apache/superset-ui/node_modules/@superset-ui/build-config/configs/typescript.js:23:22)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Can anybody help me to check the cause？
my env version is :
node.js version: v10.16.0
npm version: 6.9.0
typescript version: 3.7.3
I have merged https://github.com/apache-superset/superset-ui/pull/227

### PR DESCRIPTION
**Pull request by kristw**
🏠 Internal

* Fix #265 
* Use build configuration from `nimbus`, which is a shareable build config for `jest`, `babel`, `eslint`, `prettier`, and `typescript` on top of `beemo`. 
* Fix hundreds of new lint and ts issues. Disable some `eslint` rules for test files at the moment.

Notable overrides
* Set `timer` to `real` (`jest`'s default) instead of `nimbus`'s default which uses `fake` timer. For unknown reason, `jest.useRealTimers()` has no effect if the default `timer` is `fake`. On the other hand, if the default `timer` is `real`, `jest.useFakeTimers()` works fine. 
* Stop `eslint` from auto-fixing the `no-explicit-any` rule and convert all `any` into (the) `unknown`. 🎶❄️

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,26 +4,31 @@
   "description": "Superset UI",
   "private": true,
   "scripts": {
-    "bootstrap": "lerna bootstrap",
-    "build": "yarn run build:cjs && yarn run build:esm && yarn run type:dts && yarn run build:assets",
-    "build:cjs": "NODE_ENV=production beemo babel --extensions=\".js,.jsx,.ts,.tsx\" ./src --out-dir lib/ --delete-dir-on-start --minify --workspaces=\"@superset-ui/!(demo|generator-superset)\"",
-    "build:esm": "NODE_ENV=production beemo babel --extensions=\".js,.jsx,.ts,.tsx\" ./src --out-dir esm/ --delete-dir-on-start --esm --minify --workspaces=\"@superset-ui/!(demo|generator-superset)\"",
+    "build": "yarn run babel && yarn run type:dts && yarn run build:assets",
```
Comment (williaster) [inline package.json]: (not the most important thing but noting that) we can remove all of the `run`s with `yarn`

Comment (kristw) [reply package.json]: fixed

**Comment by kristw** [comment]
cc @etr2460 finally got a green CI

**Comment by williaster** [comment]
finally looking at this, thanks for all the effort/pain 🙏

File: packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts [hunk 2]
```diff
@@ -24,9 +24,9 @@ describe('createSmartNumberFormatter(options)', () => {
       });
       it('formats number that are >= 1 & < 1,000 as integer or float with at most 2 decimal points', () => {
         expect(formatter(1)).toBe('1');
-        expect(formatter(1.0)).toBe('1');
+        expect(formatter(1)).toBe('1');
```
Comment (williaster) [inline packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts]: ditto prettier removed the `.0`

File: packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts [hunk 3]
```diff
@@ -60,9 +60,9 @@ describe('createSmartNumberFormatter(options)', () => {
       });
       it('formats number that are >= 1 & < 1,000 as integer or float with at most 2 decimal points', () => {
         expect(formatter(-1)).toBe('-1');
-        expect(formatter(-1.0)).toBe('-1');
+        expect(formatter(-1)).toBe('-1');
```
Comment (williaster) [inline packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts]: ditto prettier removed the `.0`

File: packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts [hunk 4]
```diff
@@ -103,9 +103,9 @@ describe('createSmartNumberFormatter(options)', () => {
       });
       it('formats number that are >= 1 & < 1,000 as integer or float with at most 2 decimal points', () => {
         expect(formatter(1)).toBe('+1');
-        expect(formatter(1.0)).toBe('+1');
+        expect(formatter(1)).toBe('+1');
```
[the remark this answers is not in the document]
Comment (williaster) [reply packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts]: ohhhh, prettier removed the `.0`, maybe prettier ignore?

File: packages/superset-ui-number-format/test/factories/createSiAtMostNDigitFormatter.test.ts [hunk 5]
```diff
@@ -30,17 +30,17 @@ describe('createSiAtMostNDigitFormatter({ n })', () => {
     const formatter = createSiAtMostNDigitFormatter();
     expect(formatter(10)).toBe('10');
     expect(formatter(1)).toBe('1');
-    expect(formatter(1.0)).toBe('1');
-    expect(formatter(10.0)).toBe('10');
+    expect(formatter(1)).toBe('1');
+    expect(formatter(10)).toBe('10');
```
Comment (williaster) [inline packages/superset-ui-number-format/test/factories/createSiAtMostNDigitFormatter.test.ts]: ditto prettier removed the `.0`s  in this file

File: packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts [hunk 6]
```diff
@@ -191,15 +191,13 @@ describe('getTextDimension(input)', () => {
...
     getMultipleTextDimensions({
       texts: [SAMPLE_TEXT[0], SAMPLE_TEXT[1]],
     });
 
     expect(document.querySelectorAll('svg')).toHaveLength(1);
-
-    return promiseTimeout(() => {
-      expect(document.querySelector('svg')).toBeNull();
-    }, 600);
+    await promiseTimeout(() => {}, 501);
```
Comment (williaster) [inline packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts]: this is a case where it seems like we would want to use fake timers, otherwise it adds `0.5s` to the tests? I'm sure there were other edge cases.

File: packages/superset-ui-dimension/src/getTextDimension.ts [hunk 7]
```diff
@@ -36,7 +36,6 @@ export default function getTextDimension(
   setTimeout(() => {
     textFactory.removeFromContainer(svgNode);
     hiddenSvgFactory.removeFromContainer(container);
-    // eslint-disable-next-line no-magic-numbers
```
Comment (williaster) [inline packages/superset-ui-dimension/src/getTextDimension.ts]: wow glad this one was removed 🤦‍♂

File: packages/superset-ui-chart/test/components/createLoadableRenderer.test.tsx [hunk 8]
```diff
@@ -31,26 +34,29 @@ describe('createLoadableRenderer', () => {
...
         <LoadableRenderer onRenderSuccess={onRenderSuccess} onRenderFailure={onRenderFailure} />,
       );
       expect(loadChartSuccess).toHaveBeenCalled();
-      setTimeout(() => {
-        expect(render).toHaveBeenCalledTimes(1);
-        expect(onRenderSuccess).toHaveBeenCalledTimes(1);
-        expect(onRenderFailure).not.toHaveBeenCalled();
-        done();
-      }, 10);
+      jest.useRealTimers();
```
Comment (williaster) [inline packages/superset-ui-chart/test/components/createLoadableRenderer.test.tsx]: nice. always tricky to refactor these to not use timeouts

File: packages/superset-ui-chart/test/components/SuperChart.test.tsx [hunk 9]
```diff
@@ -33,10 +37,18 @@ describe('SuperChart', () => {
     });
   });
 
+  beforeEach(() => {
+    restoreConsole = mockConsole();
```
Comment (williaster) [inline packages/superset-ui-chart/test/components/SuperChart.test.tsx]: are these all added because nimbus would make warning throw? might be worth making comments for that since it's not obvious.

Comment (kristw) [reply packages/superset-ui-chart/test/components/SuperChart.test.tsx]: comments added

File: packages/superset-ui-chart-composition/src/ChartFrame.tsx [hunk 10]
```diff
@@ -35,8 +35,8 @@ export default class ChartFrame extends PureComponent<Props, {}> {
           }}
         >
           {renderContent({
-            height: Math.max(contentHeight || 0, height),
-            width: Math.max(contentWidth || 0, width),
+            height: Math.max(contentHeight ?? 0, height),
```
[the remark this answers is not in the document]
Comment (kristw) [reply packages/superset-ui-chart-composition/src/ChartFrame.tsx]: This sounds scary. I checked the storybook previews and they work correctly. 
What are the issues that appear in app? I could try to revert back to `||` and suppress `eslint` if it is risky.

File: packages/superset-ui-number-format/test/factories/createSiAtMostNDigitFormatter.test.ts (hunk 5 above)
Comment (kristw) [reply packages/superset-ui-number-format/test/factories/createSiAtMostNDigitFormatter.test.ts]: added `prettier-ignore`

File: packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts (hunk 2 above)
Comment (kristw) [reply packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts]: added `prettier-ignore`

File: packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kristw) [reply packages/superset-ui-number-format/test/factories/createSmartNumberFormatter.test.ts]: added `prettier-ignore`

File: packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts (hunk 6 above)
Comment (kristw) [reply packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts]: I tried but couldn't get this one to work with fake timers. The underlying logic that uses `setTimeout` has a debouncing effect + dependency tracking counter that is a bit complicated. Prefer to leave it like this for now.

Comment (kristw) [reply packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts]: I open this as a separate issue and will resolve this PR for now. https://github.com/apache-superset/superset-ui/issues/286

File: packages/superset-ui-chart-composition/src/ChartFrame.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (etr2460) [reply packages/superset-ui-chart-composition/src/ChartFrame.tsx]: you need to make sure that babel updated to handle the new syntax. babel 7.8.3 should do the trick.
